src/ferui-components/tree-view/interfaces.ts

```diff
@@ -0,0 +1,87 @@
+import { TemplateRef } from '@angular/core';
+
+/**
+ * Tree Node Data Interface
+ */
+export interface TreeNodeData<T> {
+  data: T;
+  field: string;
+}
+
+/**
+ * Paging Params Interface
+ */
+export interface PagingParams {
+  offset: number;
+  limit: number;
+}
+
+/**
+ * Tree Node Data Retriever Interface for client-side tree view
+ */
+export interface TreeNodeDataRetriever<T> {
+  // Get promise that returns TreeNodeData children array from Tree Node
+  getChildNodeData(parent: TreeNodeData<T>): Promise<Array<TreeNodeData<T>>>;
+  // Get promise that returns whether TreeNode has children nodes
+  hasChildNodes(parent: TreeNodeData<T>): Promise<boolean>;
+  // If the developer wants to give us an icon template for the expansion/collapse of node
+  getIconTemplate?(): TemplateRef<any>;
+  // If developer wishes to give us a template ref we shall render this template in the view
+  getNodeTemplate?(): TemplateRef<any>;
+}
+
+/**
+ * Paged Tree Node Data Retriever for server-side tree view
+ */
+export interface PagedTreeNodeDataRetriever<T> extends TreeNodeDataRetriever<T> {
+  getPagedChildNodeData(parent: TreeNodeData<T>, pagingParams: PagingParams): Promise<Array<TreeNodeData<T>>>;
+}
+
+/**
+ * Non root tree node empty singleton instance
+ */
+export class NonRootTreeNode implements TreeNodeData<any> {
```

Comment:
Maybe i miss something, but I'm still not sure that we need a specific interface or class to handle non-root tree nodes. Why not consider all tree nodes as non-root tree nodes and if there is only one node... Well, we're still good isn't it ?
What do you think @cfsnyder @ariquinones ? If this is mandatory i would be glad to have some explanations.

Reply:
The API that we've defined requires the user to provide the root tree node as part of the initial configuration, but there are cases where a tree view does not have a single root node. This is used to differentiate that use case.